### ISSUE TITLE
[2019] Large Batch Optimization for Deep Learning: Training BERT in 76 minutes

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#136]
## 0. 論文

| タイトル | Large Batch Optimization for Deep Learning: Training BERT in 76 minutes |
|:-------|:--|
| リンク  | https://arxiv.org/abs/1904.00962 |
| 著者    | Yang You, Jing Li, Sashank Reddi, Jonathan Hseu, Sanjiv Kumar, Srinadh Bhojanapalli, Xiaodan Song, James Demmel, Kurt Keutzer, Cho-Jui Hsieh |
| 投稿日付 | ICLR 2020 |

## 1. どんなもの？

BERT などの巨大モデルの学習高速化のために大きな batch size での学習が行われるが、汎化性能が悪化したり learning rate のチューニングが難しいことが知られている。
本研究では大きな batch size で安定した学習を行うことが出来る LAMB Optimizer を提案。
先行研究である LARS を、LARS は momentum optimizer ベースなのを Adam (second moment まで見る) ベースに変更している。(他にもある? わかってない)
結果として、性能を保ったまま巨大な batch size により BERT の学習を 76分で完了することができている。LARS よりも安定。

## 2. 先行研究と比べてどこがすごい？

LARS は task により適用できない事例があったが、LAMB は LARS でうまく行かなかった BERT の学習にも成功。

## 3. 技術や手法のキモはどこ？

- LARS を Adam のような second momentum まで見るように変更
    - ![スクリーンショット 2021-07-09 14 07 59](https://user-images.githubusercontent.com/1255485/125026235-19bb1b00-e0bf-11eb-8e20-ebe4cfb3e221.png)
- LARS/LAMB での学習時の収束について upper bound を導入

## 4. どうやって有効だと検証した？

BERT の事前学習にて batch size を大きくして検証。LARS をはじめとする他の optimizer が大きな batch size で性能が落ちるのに対して、高いスコアを維持した(スコアは事前学習後に SQuAD v1 で検証)。

![スクリーンショット 2021-07-09 14 09 22](https://user-images.githubusercontent.com/1255485/125026364-5d158980-e0bf-11eb-9ba0-4c241f50146b.png)
![スクリーンショット 2021-07-09 14 09 47](https://user-images.githubusercontent.com/1255485/125026369-60107a00-e0bf-11eb-9170-46403e86796d.png)

## 6. 次に読むべき論文は？

- Large Batch Training of Convolutional Networks
    - https://arxiv.org/abs/1708.03888v3
    - LARS の提案論文
